### PR TITLE
Add rollback test for linux-bridge

### PR DESCRIPTION
**Pull request by qinqon**
To verify #516

Signed-off-by: Quique Llorente <ellorent@redhat.com>

### CODE REVIEW

File: tests/integration/linux_bridge_test.py [hunk 1]
```diff
@@ -295,6 +297,19 @@ def test_linux_bridge_over_bond_over_slave_in_one_transaction(bond0):
...
+    current_state = libnmstate.show()
+    bridge_name = TEST_BRIDGE0
+    bridge_state = _create_bridge_subtree_config(("invalid_port",))
+    with pytest.raises(NmstateVerificationError):
+        with linux_bridge(bridge_name, bridge_state) as desired_state:
+            libnmstate.apply(desired_state)
+
+    time.sleep(5)  # Give some time for NetworkManager to rollback
+    current_state_after_apply = libnmstate.show()
+    assert current_state == current_state_after_apply
```
Comment (phoracek) [inline tests/integration/linux_bridge_test.py]: It's not current state anymore. Please rename it to `original_state` or something alike.

**Comment by cathay4t** [comment]
@qinqon I tried to reproduce it in RHEL 8.1 and seems works well. And the CI also show the works well.

Which  system(Fedora or RHEL) are you using to reproduce the problem?

The `device is strictly unmanaged` issue is fixed in NM 1.20 which was related to the race between NM and udev.

File: tests/integration/linux_bridge_test.py [hunk 2]
```diff
@@ -295,6 +297,19 @@ def test_linux_bridge_over_bond_over_slave_in_one_transaction(bond0):
...
+    original_state = libnmstate.show()
+    bridge_name = TEST_BRIDGE0
+    bridge_state = _create_bridge_subtree_config(("invalid_port",))
+    with pytest.raises(NmstateVerificationError):
+        with linux_bridge(bridge_name, bridge_state) as desired_state:
+            libnmstate.apply(desired_state)
+
+    time.sleep(5)  # Give some time for NetworkManager to rollback
+    current_state = libnmstate.show()
+    assert original_state == current_state
```
Comment (cathay4t) [inline tests/integration/linux_bridge_test.py]: In the issue, you mentioned there is leftover bridge interface. Please consider to add a check on output of `ip link show <br_name>` to confirm there is no leftover bridge.

Comment (EdDev) [reply tests/integration/linux_bridge_test.py]: We are supposed to report *all* interfaces, even if there are no profiles for it. (they will just show up as `DOWN`)
Therefore, using `ip link` should not be needed.

File: tests/integration/linux_bridge_test.py [hunk 3]
```diff
@@ -295,6 +297,19 @@ def test_linux_bridge_over_bond_over_slave_in_one_transaction(bond0):
     assertlib.assert_absent(bridge_name)
 
 
+def test_rollback_for_linux_bridge():
+    original_state = libnmstate.show()
+    bridge_name = TEST_BRIDGE0
+    bridge_state = _create_bridge_subtree_config(("invalid_port",))
```
Comment (EdDev) [inline tests/integration/linux_bridge_test.py]: The invalid port name is confusing and may fail the transaction before anything was created (which will not check what you intended to check).
Please add a key-value that is invalid at the `desired_state` end result, just before you apply.

BTW: please always use single quotes (`).

Comment (qinqon) [reply tests/integration/linux_bridge_test.py]: Done

File: tests/integration/linux_bridge_test.py (hunk 2 above)
Comment (phoracek) [reply tests/integration/linux_bridge_test.py]: That is the issue of the bug we've found AFAIK. The interface is not reported in state after rollback, but it is in kernel. (But not 100% sure here)

https://github.com/nmstate/nmstate/issues/516

Comment (EdDev) [reply tests/integration/linux_bridge_test.py]: Where did you see that the interface is not reported? I do not see it in the issue description.

Comment (phoracek) [reply tests/integration/linux_bridge_test.py]: I'm not sure actually. nmstate was not able to remove the interface from system, but I don't know whether it was reported in `show` or not.

**Comment by EdDev** [comment]
@qinqon , you merged instead of rebasing. Please rebase over master such that this PR will include only you changes and no merge commits.

**Comment by qinqon** [comment]
@EdDev rebased

**Comment by EdDev** [comment]
@qinqon , any reason you reseted my ack on this pr? I did not see any changes since the last ack.

**Comment by qinqon** [comment]
@EdDev nop, all good, travis failure is unrelated I think.

**Comment by EdDev** [comment]
@qinqon , please rebase.

**Pull request merged**